backend/de.metas.contracts/src/main/java/de/metas/contracts/bpartner/process/C_BPartner_DeactivateAfterOrgChange.java

```diff
@@ -0,0 +1,135 @@
+				.andCollectChildren(I_AD_OrgChange_History.COLUMNNAME_C_BPartner_From_ID, I_C_BPartner.class)
+				.create()
+				.listIds();
+
+		if(Check.isEmpty(partnerIdsToDeactivate))
+		{
+			// nothing to do
+			return;
+		}
+		deactivateBankAccounts(partnerIdsToDeactivate, today);
+		deactivateUsers(partnerIdsToDeactivate, today);
+		deactivateLocations(partnerIdsToDeactivate, today);
+		deactivatePartners(partnerIdsToDeactivate, today);
+
+	}
+	private void deactivateBankAccounts(final List<Integer> partnerIds, final LocalDate date)
+	{
+		final ICompositeQueryUpdater<I_C_BP_BankAccount> queryUpdater = queryBL.createCompositeQueryUpdater(I_C_BP_BankAccount.class)
+				.addSetColumnValue(I_C_BP_BankAccount.COLUMNNAME_IsActive, false)
+				.addSetColumnValue(I_C_BP_BankAccount.COLUMNNAME_Updated, date)
+				.addSetColumnValue(I_C_BP_BankAccount.COLUMNNAME_UpdatedBy, Env.getAD_User_ID());
+		queryBL
+				.createQueryBuilder(I_C_BP_BankAccount.class)
+				.addInArrayFilter(I_C_BP_BankAccount.COLUMNNAME_C_BPartner_ID, partnerIds)
+				.create()
+				.updateDirectly(queryUpdater);
+	}
+	private void deactivateUsers(final List<Integer> partnerIds, final LocalDate date)
+	{
+		final ICompositeQueryUpdater<I_AD_User> queryUpdater = queryBL.createCompositeQueryUpdater(I_AD_User.class)
+				.addSetColumnValue(I_AD_User.COLUMNNAME_IsActive, false)
+				.addSetColumnValue(I_AD_User.COLUMNNAME_Updated, date)
+				.addSetColumnValue(I_AD_User.COLUMNNAME_UpdatedBy, Env.getAD_User_ID());
+		queryBL
+				.createQueryBuilder(I_AD_User.class)
+				.addInArrayFilter(I_AD_User.COLUMNNAME_C_BPartner_ID, partnerIds)
+				.create()
+				.updateDirectly(queryUpdater);
+	}
+
+	private void deactivateLocations(final List<Integer> partnerIds, final LocalDate date)
+	{
+		final ICompositeQueryUpdater<I_C_BPartner_Location> queryUpdater = queryBL.createCompositeQueryUpdater(I_C_BPartner_Location.class)
+				.addSetColumnValue(I_C_BPartner_Location.COLUMNNAME_IsActive, false)
+				.addSetColumnValue(I_C_BPartner_Location.COLUMNNAME_Updated, date)
+				.addSetColumnValue(I_C_BPartner_Location.COLUMNNAME_UpdatedBy, Env.getAD_User_ID());
+		queryBL
+				.createQueryBuilder(I_C_BPartner_Location.class)
+				.addInArrayFilter(I_C_BPartner_Location.COLUMNNAME_C_BPartner_ID, partnerIds)
+				.create()
+				.updateDirectly(queryUpdater);
+	}
+
+
+	private void deactivatePartners(final List<Integer> partnerIds, final LocalDate date)
+	{
+		final ICompositeQueryUpdater<I_C_BPartner> queryUpdater = queryBL.createCompositeQueryUpdater(I_C_BPartner.class)
+				.addSetColumnValue(I_C_BPartner.COLUMNNAME_IsActive, false)
+				.addSetColumnValue(I_C_BPartner.COLUMNNAME_Updated, date)
+				.addSetColumnValue(I_C_BPartner.COLUMNNAME_UpdatedBy, Env.getAD_User_ID());
+		queryBL
+				.createQueryBuilder(I_C_BPartner.class)
+				.addInArrayFilter(I_C_BPartner.COLUMNNAME_C_BPartner_ID, partnerIds)
+				.create()
+				.updateDirectly(queryUpdater);
```

Comment:
for the bpartners i suggest you even iterate them and do `addLog` so they end up in the AD_PInstance-Log

Reply:
OK like this? 

```java
		partnerIdsToDeactivate.stream()
				.forEach(
						partnerId -> addLog("Business Partner {} was deactivated because it was moved to another organization.",
											partnerId));
											
```

Or do we have a standard?